src/services/codefixes/importFixes.ts

```diff
@@ -501,6 +501,14 @@ namespace ts.codefix {
                             relativeFileName = getRelativePath(moduleFileName, sourceDirectory);
                         }
 
+                        if (startsWith(relativeFileName, "@types/")) {
+                            relativeFileName = relativeFileName.substr(/*"@types".length*/ 7);

```

Comment:
`"@types".length` is 6. Why not just use that expression? You could even add a `stripStart(str, prefix): string | undefined` helper so you don't have to repeat this kind of code.